thor_devkit/transaction.py

```diff
@@ -303,23 +423,50 @@ def encode(self):
             temp.update({
                 'signature': self.signature
             })
-            return ComplexCodec(SignedTxWrapper).encode(temp)
+            if self.get_type() == TransactionType.DYNAMIC_FEE:
```

Comment:
Rework the logic to have just singular return statement, not 4 possible returns.